File: cpp_solutions/1863.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int n = nums.size();
        int res = 0;
        for (int i = 0; i < (1 << n); ++i) {
            int temp = 0;
            for (int j = 0; j < n; ++j) {
                if (i & (1 << j)) temp ^= nums[j];
            }
            res += temp;
        }
        return res;
    }
};

File: rust_solutions/1863.rs
impl Solution {
    pub fn subset_xor_sum(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut res = 0;
        for i in 0..(1 << n) {
            let mut temp = 0;
            for j in 0..n {
                if (i & (1 << j)) > 0 {
                    temp ^= nums[j];
                }
            }
            res += temp;
        }
        return res;
    }
}
